[CLOCK] Modified date & middle digits displays

JackZ974 opened the pull request:
<!-- ⚠⚠ Do not delete this pull request template! ⚠⚠ -->
<!-- Pull requests that do not follow this template are likely to be ignored. -->
[CLOCK] modified date format & display of seconds/year
<!-- Add the issues this PR fixes here. If no issues are related to this PR, then this line can be removed. -->
Fixes #[issue_no]

**Summary of Changes**
<!-- Please provide a summary of changes for this pull request, ensuring all changes are explained. -->

    Added the extraction of Seconds and Year from the Time string in order to display it with a smaller font as IRL
    The display of date is now MON/DAY/YEAR as expected
    removed the display of ":" in the date display


**Screenshots (if necessary)**
<!-- If your PR includes visual changes, screenshots from before and after your change should always be included. -->
Before:
![image](https://user-images.githubusercontent.com/23705479/93238682-6decbb80-f78a-11ea-9700-4b75b776fca7.png)
After:
![image](https://user-images.githubusercontent.com/23705479/93238721-7b09aa80-f78a-11ea-928f-c1f1d869f2c6.png)
Date display:
![image](https://user-images.githubusercontent.com/23705479/93238765-8b218a00-f78a-11ea-8c21-a31f30332f3d.png)
Real one:
![image](https://user-images.githubusercontent.com/23705479/93238817-9c6a9680-f78a-11ea-81d5-89b375ec57aa.png)


**Additional context**
<!-- Add any other context about the pull request here. -->

<!-- You may optionally provide your discord username, so that we may contact you directly about the issue. -->
Discord username:

camilo625 commented:
Works as described. I am seeing CHR, UTC and ET in the textures (really blurry), not shure if that's normal.

JackZ974 commented:
> 
> 
> Works as described. I am seeing CHR, UTC and ET in the textures (really blurry), not shure if that's normal.

The texture is not mine, but considering the size of the overall texture file, not so much details can be added, I'm afraid. AFAIK, even the small blue legends below the white one were unreadable and couldn't make it to the texture

veikkos commented:
I'm confused. Why doesn't the "before" image have the CHR/UTC/ET texts like there currently is with master? Have you removed them? But not in this PR?

JackZ974 commented:
Simply because, the rate of PR is going so fast in this project that I gave up on constanatly updating my sim when I'm coding. So i still have the old texture without the texts

AdenFlorian reviewed (approved):
ready to test! (it needs a retest because the code changed since last test)

JackZ974 commented:
Minor addition in the css of a dark colored background that lights up at night. 
![image](https://user-images.githubusercontent.com/23705479/93279546-33a60d00-f7d0-11ea-939b-d3220824e84c.png)

starvingjavi commented:
Tested during the day and the brightness of the numbers is not uniform between them like the reference pictures. During the night the brightness difference is not so apparent but it is still there
![Captura de pantalla 2020-09-15 201813](https://user-images.githubusercontent.com/70245523/93281197-37666f00-f791-11ea-8da0-ab119f99557d.png)
![Captura de pantalla 2020-09-15 201838](https://user-images.githubusercontent.com/70245523/93281198-37ff0580-f791-11ea-9e33-bc455dbfa4b3.png)

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css:
```diff
@@ -51,34 +51,55 @@ a320-neo-clock-element {
   overflow: hidden; }
   a320-neo-clock-element #Mainframe {
     position: absolute;
+    /* top: -4%; */
```
Please remove comments ;)

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css:
```diff
@@ -51,34 +51,55 @@ a320-neo-clock-element {
+    color: rgba(155, 148, 148, 0.911);}
     a320-neo-clock-element #Mainframe #TopSelector {
       margin-top: 0%;
+      /* width: 100%;
+      height: 35%; */
```
Please remove comments ;)

JackZ974 replied:
Done

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.css:
```diff
@@ -51,34 +51,55 @@ a320-neo-clock-element {
+        margin-right: 5%;
+        width: 95%;
+        text-align: right;
+        /* font-kerning: none; */
```
Please remove comments ;)

JackZ974 replied:
I will leave this one for now as a placeholder. This waiting for a new monospaced font to be creted for the chrono. #fonts guys are on it as it seems

Benjozork replied:
> I will leave this one for now as a placeholder. This waiting for a new monospaced font to be created for the chrono. #fonts guys are on it as it seems
> 

Okay perfect :)

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
-                const currentDisplayTime = SimVar.GetSimVarValue("L:PUSH_CHRONO_SET", "Bool") ? this.getUTCDate() : this.getUTCTime()
+                let currentDisplayTime = "";
+                let currentDisplayTime2 = "";
+                if (SimVar.GetSimVarValue("L:PUSH_CHRONO_SET", "Bool") == true) {
```
```suggestion
                if (SimVar.GetSimVarValue("L:PUSH_CHRONO_SET", "Bool") === 1) {
```

`== true`

JackZ974 replied:
triple === ?

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
+                let currentDisplayTime = "";
+                let currentDisplayTime2 = "";
+                if (SimVar.GetSimVarValue("L:PUSH_CHRONO_SET", "Bool") == true) {
+
```
```suggestion
```

JackZ974 replied:
This was from a previous PR, not my part of code, but OK

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
+
+                    currentDisplayTime = this.getUTCDate(); 
+                    currentDisplayTime2 = this.getUTCYear(); 
+                    
```
```suggestion
```

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -57,10 +60,25 @@ class A320_Neo_Clock extends BaseAirliners {
                 if (currentDisplayTime !== this.lastDisplayTime) {
                     this.lastDisplayTime = currentDisplayTime;
                     this.middleSelectorElem.textContent = currentDisplayTime;
+                    
```
```suggestion
```

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -78,25 +96,36 @@ class A320_Neo_Clock extends BaseAirliners {
-        const Year = `${SimVar.GetGlobalVarValue("ZULU YEAR", "number")}`.substr(2,4)
-
-        return `${Day}.${Month}.${Year}`
+        const Day = SimVar.GetGlobalVarValue("ZULU DAY OF MONTH", "number");
+        const Month = `${SimVar.GetGlobalVarValue("ZULU MONTH OF YEAR", "number")}`.padStart(2,'0');
+        return `${Month}.${Day}`;
```
```suggestion
        const day = SimVar.GetGlobalVarValue("ZULU DAY OF MONTH", "number");
        const month = `${SimVar.GetGlobalVarValue("ZULU MONTH OF YEAR", "number")}`.padStart(2,'0');
        return `${month}.${day}`;
```

JackZ974 replied:
Was like this in the original ASOBO code

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -78,25 +96,36 @@ class A320_Neo_Clock extends BaseAirliners {
     }
 
+    getUTCYear() {
+        const Year = `${SimVar.GetGlobalVarValue("ZULU YEAR", "number")}`.substr(2,4);
+        return `${Year}`;
```
```suggestion
        const year = `${SimVar.GetGlobalVarValue("ZULU YEAR", "number")}`.substr(2,4);
        return `${year}`;
```

JackZ974 replied:
this var was labelled like this in the original ASOBO code

Benjozork commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -6,15 +6,18 @@ class A320_Neo_Clock extends BaseAirliners {
         this.lastFlightTime = null;
         this.lastLocalTime = 0;
         this.lastResetVal = null;
+
```
```suggestion
```

Benjozork reviewed (commented):
Thank you for your PR, and nice use of `const` everywhere possible :)

Few things to correct but looking good ! I put suggestions on each so you can quickly fix it :)

JackZ974 commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -78,25 +96,36 @@ class A320_Neo_Clock extends BaseAirliners {
         return "";
     }
 
+    getUTCSeconds() {
+        const value = SimVar.GetGlobalVarValue("ZULU TIME", "seconds");
+        if (value) {
+            const seconds = Number.parseInt(value);
+            const time = Utils.SecondsToDisplayTime(seconds, true, true, false);
+            return time.toString().substr(6,2);
+        }
+        return "";
+    }
```
Good point. I'll look into that

JackZ974 replied:
Done the modification. Thx for pointing this out

JackZ974 commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/Clock/A320_Neo_Clock.js:
```diff
@@ -78,25 +96,36 @@ class A320_Neo_Clock extends BaseAirliners {
     }
 
+    getUTCYear() {
+        const Year = `${SimVar.GetGlobalVarValue("ZULU YEAR", "number")}`.substr(2,4);
+        return `${Year}`;
```
Ok. not so familiar with the subtleties of js coding. Always interesting to learn from the js Masters

JackZ974 commented:
> 
> 
> Tested during the day and the brightness of the numbers is not uniform between them like the reference pictures. During the night the brightness difference is not so apparent but it is still there
> ![Captura de pantalla 2020-09-15 201813](https://user-images.githubusercontent.com/70245523/93281197-37666f00-f791-11ea-8da0-ab119f99557d.png)
> ![Captura de pantalla 2020-09-15 201838](https://user-images.githubusercontent.com/70245523/93281198-37ff0580-f791-11ea-9e33-bc455dbfa4b3.png)
> 
> Tried changing the brightness knobs to see if it would change anything and nothing, it stays the same
You probably had the early version of the css
i modified the css in the meantime for the background and corrected this issue as well. Would you mind testing it again? This should be ok now

JackZ974 commented:
![image](https://user-images.githubusercontent.com/23705479/93479799-6a317400-f905-11ea-8c0c-aadeb16fd42a.png)

AdenFlorian reviewed (approved):
ready to test again!

UncleClapton reviewed (approved):
LGTM!

ImenesFBW commented:
Works and looks as described. It's really sleek and adds a lot to cockpit immersion. Also you can see the min markers clearly from the pilot seat now.  Can't wait to see this in the master thanks for your work!

Now this might be unrelated but will add it here for future reference. Had to click the cabin ALL a few times for ecam to register it off the C/L.
